### ISSUE TITLE
SemanticDB occurrence for default contructor should be zero length

### ISSUE THREAD

**Issue by ckipp01** [scalameta/metals#2051]
**Describe the bug**
When attempting a goto definition on the type of a parameter in a class, you are brought to the site(s) where the class is created instead of the expected type definition.

```scala
class Test(a: In@@t)

val a = new Test(1)
val b = new Test(2)
```

In the above example, I'd expect to be brought to the `Int` class. This is what happens with Scala 2. However, in Dotty, this example will return the two `Test` creation sites.

```
[Trace - 02:08:29 PM] Sending response 'textDocument/definition - (3)'. Processing request took 42ms
Result: [
  {
    "uri": "file:///Users/ckipp/Documents/scala-workspace/scalameta-org/dotty-implement-all/src/main/scala/Main.scala",
    "range": {
      "start": {
        "line": 15,
        "character": 18
      },
      "end": {
        "line": 15,
        "character": 18
      }
    }
  },
  {
    "uri": "file:///Users/ckipp/Documents/scala-workspace/scalameta-org/dotty-implement-all/src/main/scala/Main.scala",
    "range": {
      "start": {
        "line": 16,
        "character": 18
      },
      "end": {
        "line": 16,
        "character": 18
      }
    }
  }
]
```

**To Reproduce**
1. Copy the code above or just attempt a goto definition on a parameter type in a class.
2. See where you end up.

**Expected behavior**
I'd expect to go to the definition site of the type.

**Installation:**
- Operating system: macOS
- Editor: Vim
- Metals version: `v0.9.3+65-02770a08-SNAPSHOT`

**Additional context**
Note that when the same thing is done with a method, you are correctly brought to the type definition site.

```scala
def test(a: In@@t)

test(1)
test(2)
```

This works as expected, and I'd expect the above to do the same.

**Search terms**
Dotty, goto definition

**Comment by tgodzik** [comment]
Thanks for reporting! That might have something to do with semanticdb generation in Dotty. I think the position might be wrong for test and encompass the params.

**Comment by tgodzik** [comment]
Opened an issue in Dotty and we already had a look: https://github.com/lampepfl/dotty/issues/9727

**Comment by dos65** [comment]
Can't reproduce anymore. Fixed in lampepfl/dotty#9768
